api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsService.java

```diff
@@ -722,22 +722,22 @@ private static Map<Long, Long> getCreatorByWorkspaceCache(
   @NotNull
   private static Map<Long, Double> getDbCostByCreatorCache(
       List<WorkspaceCostView> allCostsInDbForUsers) {
-    final Map<Long, Double> dbCostByCreator =
-        allCostsInDbForUsers.stream()
+    return allCostsInDbForUsers.stream()
             .collect(
                 Collectors.groupingBy(
                     WorkspaceCostView::getCreatorId,
                     Collectors.summingDouble(
                         v -> Optional.ofNullable(v.getInitialCreditsCost()).orElse(0.0))));
-    return dbCostByCreator;
   }
 
-  private List<Workspace> getWorkspacesForUser(DbUser user) {
-    return workspaceMapper
-        .toApiWorkspaceResponseList(workspaceDao, fireCloudService.listWorkspacesAsService(), this)
-        .stream()
-        .map(WorkspaceResponse::getWorkspace)
-        .filter(ws -> ws.getCreatorUser().getUserName().equals(user.getUsername()))
-        .collect(Collectors.toList());
+  private List<DbWorkspace> getWorkspacesForUser(DbUser user) {
```

Comment:
Swapped function to use internal dao instead. I am wondering if it is worth shifting `workspaceDao.findAllByCreator(user)` into a WorkspaceService function.